frontend/src/metabase/query_builder/components/GuiQueryEditor.jsx

```diff
@@ -162,10 +196,10 @@ export default class GuiQueryEditor extends Component {
                     <AggregationWidget
                         key={"agg"+index}
                         aggregation={aggregation}
-                        tableMetadata={tableMetadata}
-                        customFields={Query.getExpressions(this.props.datasetQuery.query)}
-                        updateAggregation={(aggregation) => this.props.updateQueryAggregation(index, aggregation)}
-                        removeAggregation={canRemoveAggregation ? this.props.removeQueryAggregation.bind(null, index) : null}
+                        tableMetadata={query.tableMetadata()}
```

Comment:
Why does the aggregation widget need to get the table metadata. Code smell that we haven't thought through aggregations/filters/breakouts enough.